python/ray/data/dataset.py

```diff
@@ -521,13 +521,24 @@ def transform(
             *fn_args,
             **fn_kwargs,
         ) -> Iterable[Block]:
+
+            # Fetch next available input blocks for the batch.
+            # Only fetch required number of blocks but not all blocks.
+            # This avoids over-buffering in current node's memory.
+            blocks_iter = iter(blocks)
+            def _fetch_next_blocks(batcher: Batcher, batch_size: Optional[int]):
+                for block in blocks_iter:
+                    batcher.add(block)
+                    if batch_size is None or batcher.buffer_size() >= batch_size:
```

Comment:
```suggestion
                    if batch_size is not None and batcher.buffer_size() >= batch_size:
```

I think it should be this? (took the inverse of the other if check)

Reply:
if `batch_size` is None, we actually treat each block as a batch. So We only need to get one block for next batch.

Reply:
Ah I see, thanks! Maybe add a comment to this effect?

Reply:
Reusing the existing code path, so this new change is no longer needed. But I agree with the suggestion :)